concrete/controllers/single_page/dashboard/system/registration/automated_logout.php

```diff
@@ -0,0 +1,135 @@
+use Concrete\Core\Http\ResponseFactory;
+use Concrete\Core\Page\Controller\DashboardPageController;
+use Concrete\Core\Page\Page;
+use Concrete\Core\Url\Resolver\Manager\ResolverManager;
+
+class AutomatedLogout extends DashboardPageController
+{
+
+    const ITEM_IP = 'concrete.security.session.invalidate_on_ip_mismatch';
+    const ITEM_USER_AGENT = 'concrete.security.session.invalidate_on_user_agent_mismatch';
+    const ITEM_SESSION_INVALIDATE = 'concrete.session.valid_since';
+
+    /**
+     * The response factory we use to generate responses
+     *
+     * @var ResponseFactory
+     */
+    protected $factory;
+
+    /**
+     * The config repsository we save to
+     *
+     * @var \Concrete\Core\Config\Repository\Repository
+     */
+    protected $config;
+
+    /**
+     * The URL manager that tells us how to get to the page we're looking for
+     *
+     * @var \Concrete\Core\Url\Resolver\Manager\ResolverManager
+     */
+    protected $urls;
+
+    public function __construct(Page $c, ResponseFactory $factory, Repository $config, ResolverManager $urls)
+    {
+        parent::__construct($c);
+
+        $this->factory = $factory;
+        $this->config = $config;
+        $this->urls = $urls;
+    }
+
+    /**
+     * Main view function for the controller. This method is called when no other action matches the request
+     *
+     * @return \Concrete\Core\Http\Response|void
+     */
+    public function view()
+    {
+        $this->set('trustedProxyUrl', $this->urls->resolve(['/dashboard/system/permissions/trusted_proxies']));
+        $this->set('invalidateOnIPMismatch', $this->config->get(self::ITEM_IP));
+        $this->set('invalidateOnUserAgentMismatch', $this->config->get(self::ITEM_USER_AGENT));
+
+        $this->set('saveAction', $this->action('save'));
+        $this->set('invalidateAction', $this->action('invalidate_sessions', $this->token->generate('invalidate_sessions')));
+    }
+
+    /**
+     * An action for saving the Session Security form
+     * This method will manage saving settings and redirecting to the appropriate results page
+     *
+     * @return \Concrete\Core\Http\ResponseFactory|\Symfony\Component\HttpFoundation\Response
+     */
+    public function save()
+    {
+        if (!$this->token->validate('save_automated_logout')) {
+            $this->error->add($this->token->getErrorMessage());
+
+            $this->flash('error', $this->error);
+
+            // Redirect away from this save endpoint
+            return $this->factory->redirect($this->action());
+        }
+
+        $post = $this->request->request;
+
+        // Save the posted settings
+        $this->config->save(self::ITEM_IP, filter_var($post->get('invalidateOnIPMismatch'), FILTER_VALIDATE_BOOLEAN));
+        $this->config->save(self::ITEM_USER_AGENT, filter_var($post->get('invalidateOnUserAgentMismatch'), FILTER_VALIDATE_BOOLEAN));
+
+        $this->flash('message', t('Successfully saved Session Security settings'));
```

Comment:
Do we need a custom message for this? All these messages would need to be translated, whereas it doesn't add much value (imo). E.g. why not 'Settings saved successfully.'?

Reply:
I think having specific actions like `Save Security Settings` instead of things like `Save` make a lot of sense, and I'd argue that logic extends to success and error messages. If we just had a generic message everywhere it's really easy to get lost and lose track of what you're actually doing when you're in the thick of things. Having clear text instead of generic text makes it simple to get your bearings after switching back to a tab or after filling out a lengthy form, and can help prevent an overworked zoning out intern from clicking down the wrong path when configuring a site.

Reply:
A problem is that many messages are not consistent. See for example these translations:

![afbeelding](https://user-images.githubusercontent.com/1431100/47839234-4c5ed280-ddb2-11e8-84c2-4f91e3ec01fe.png)

- Some use Capitalized Words
- Some end with a dot, others not.
- The order of words differs.

![afbeelding](https://user-images.githubusercontent.com/1431100/47839324-aeb7d300-ddb2-11e8-9a74-925083743ba0.png)

- Duplication within various messages.

The translation status of a lot of languages is not complete. I think its because it's just a lot of work. https://translate.concrete5.org/translate/package/concrete5/dev-8

I hope this gives a bit of insight which might help future commits. I don't disagree with you but I think this kind of feedback helps to get the overall picture.